### PR TITLE
prevent duplicated gui hooks during registration

### PR DESCRIPTION
**Pull request by dinhlongviolin1**

### CODE REVIEW

File: taipy/gui/hook.py [hunk 1]
```diff
@@ -14,6 +14,10 @@ def __init__(self):
         self.__hooks: t.List[Hook] = []
 
     def _register_hook(self, hook: Hook):
+        # Prevent duplicated hooks
+        for h in self.__hooks:
+            if isinstance(hook, type(h)):
```
Comment (FredLL-Avaiga) [inline taipy/gui/hook.py]: what if hook is of a subclass of h ?
Is it possible ?
should we test type(hook) is type(h) ?
Should a warning message be added ?

**Review by FredLL-Avaiga — DISMISSED** [review]

File: taipy/gui/hook.py (hunk 1 above)
Comment (FabienLelaquais) [reply taipy/gui/hook.py]: Indeed I would suspect (although I really cannot see the use case at this point) that having two different hooks registered, one inheriting from the other... could happen.

**Review by FabienLelaquais — DISMISSED** [review]
I'm good with that.

File: taipy/gui/hook.py (hunk 1 above)
Comment (dinhlongviolin1) [reply taipy/gui/hook.py]: We can test the exact type by getting by comparing str(type(h)) == str(type(hook)). That would prevent the inheritance issue.

I will add a little warning as recommended

Comment (FredLL-Avaiga) [reply taipy/gui/hook.py]: why not `type(hook) is type(h)` ?

Comment (dinhlongviolin1) [reply taipy/gui/hook.py]: yes that is what I did :D

**Review by FabienLelaquais — COMMENTED** [review]
👍

**Review by FabienLelaquais — APPROVED** [review]
👍👍👍

**Pull request merged**